Form/Type/VichImageType.php

```diff
@@ -42,8 +47,18 @@ public function configureOptions(OptionsResolver $resolver): void
         $resolver->setDefaults([
             'image_uri' => true,
             'imagine_pattern' => null,
+            'storage_resolve_method' => static::STORAGE_RESOLVE_URI,
         ]);
 
+	    $resolver->setAllowedValues(
+		    'storage_resolve_method',
+		    [
+			    static::STORAGE_RESOLVE_URI,
+			    static::STORAGE_RESOLVE_PATH_RELATIVE,
+			    static::STORAGE_RESOLVE_PATH_ABSOLUTE,
+		    ]
+	    );
+
         $resolver->setAllowedTypes('image_uri', ['bool', 'string', 'callable']);
 
         $imageUriNormalizer = static function (Options $options, $imageUri) {
```

Comment:
@garak I'm not sure about the return type here. It can be a bool, a string or even a callable, however PHP does not support multiple return type declarations (yet) - or at least I'm not aware of such. "Mixed" is not an option. Some people suggest to use an Interface for this, but that seems to be an overkill to me...

What are your thoughts? We have a very similar normalizer function for **_VichFileType_** class. Maybe this type hinting job should be done in a separate PR to finally release the solution for the original problem.

Reply:
Missing return type for mixed cases is fine